### PR TITLE
config: Add from_dict method

### PR DESCRIPTION
**Pull request by chivay**
Now you can 
```python
from karton.core.Config import Config
config = Config.from_dict({
    "minio": {...},
    "redis":  {...},
})
```

It's ugly, but it works!

### CODE REVIEW

File: karton/core/config.py [hunk 1]
```diff
@@ -44,6 +44,20 @@ def __init__(self, path: Optional[str] = None) -> None:
...
 
+    @staticmethod
+    def from_dict(config_dict):
+        # Create the object. This will behave in a normal way, by scaning
+        # the filesystem. If it fails, RuntimeError will be raised.
+        # We can ignore it. Created ConfigParser will be overwritten by ours
+        try:
+            config = Config()
+        except RuntimeError:
+            pass
```
Comment (msm-code) [inline karton/core/config.py:55]: What's the point of this :thinking:.

Config won't even be defined if RuntimeError is raised. Try:

```
>>> try:
...     kot = 1/0
... except:
...     pass
...
>>> print(kot)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'kot' is not defined
```

Unless I'm wrong and somehow half-initialised objects can exist in Python?

**Review by msm-code — CHANGES_REQUESTED** [review]
i think try/except needs a deeper look

**Comment by chivay** [comment]
ah, I'm dumb

File: karton/core/config.py [hunk 2]
```diff
@@ -39,10 +43,18 @@ def __init__(self, path: Optional[str] = None) -> None:
...
-        if not self.config.has_section("redis"):
-            raise RuntimeError("Missing Redis configuration")
+        if check_sections:
+            if not self.config.has_section("minio"):
+                raise RuntimeError("Missing MinIO configuration")
+            if not self.config.has_section("redis"):
+                raise RuntimeError("Missing Redis configuration")
+
+    @staticmethod
+    def from_dict(config_dict):
```
Comment (msm-code) [inline karton/core/config.py:53]: maybe?

```suggestion
    def from_dict(config_dict: Dict) -> Config:
```

(this will need additional import on the top)

**Review by msm-code — APPROVED** [review]
LGTM

File: karton/core/config.py (hunk 2 above)
Comment (chivay) [reply karton/core/config.py]: and also `"Config"` instead of `Config`.  Applied.

**Comment by msm-code** [comment]
:shipit:

**Pull request merged**